components/script/dom/performance.rs

```diff
@@ -273,10 +298,31 @@ impl Performance {
             o.notify();
         }
     }
-
     fn now(&self) -> f64 {
         (time::precise_time_ns() - self.navigation_start_precise).to_ms()
     }
+    fn can_add_resource_timing_entry(&self) -> bool {
+		if self.resource_timing_buffer_current_size<=self.resource_timing_buffer_size_limit{
+			return true;
+		}
+		else {
+			return false;
+			}
+}
+	fn copy_secondary_resource_timing_buffer(&self) {
+		while self.can_add_resource_timing_entry() && self.resource_timing_secondary_entries.borrow_mut().len()>0 {
+		self.queue_entry(&(*(self.resource_timing_secondary_entries.borrow_mut().pop_front().unwrap())),true);
+		}
+	}
+	fn fire_buffer_full_event(&self) {
```

Comment:
We are missing a few steps from the spec here (i.e. steps 1.1, 1.4, 1.5). Please, try to follow the spec as much as possible.